test/unittest/tests/block/test_block_api_write.cpp

```diff
@@ -0,0 +1,216 @@
+    if (data_checksum)
+        CHECK(blk->flags == WT_BLOCK_DATA_CKSUM);
+    else
+        CHECK(blk->flags == 0);
+    __wt_buf_free(nullptr, &read_buf);
+}
+
+// Test that previous write buffers are still present in the block manager.
+void
+test_validate_cookies(WT_BM *bm, std::shared_ptr<mock_session> session,
+  const std::vector<std::pair<std::array<uint8_t, WT_BTREE_MAX_ADDR_COOKIE>, size_t>> &cookies,
+  const std::vector<std::string> &expected_strings)
+{
+    for (int i = 0; i < cookies.size(); i++) {
+        const auto &cookie = cookies.at(i);
+
+        // Make sure that the cookie is valid still.
+        REQUIRE(bm->addr_invalid(
+                  bm, session->get_wt_session_impl(), cookie.first.data(), cookie.second) == 0);
+
+        WT_ITEM read_buf;
+        WT_CLEAR(read_buf);
+        REQUIRE(bm->read(bm, session->get_wt_session_impl(), &read_buf, cookie.first.data(),
+                  cookie.second) == 0);
+
+        // The data should match the expected string.
+        const auto &str = expected_strings.at(i);
+        CHECK(memcmp(str.data(), WT_BLOCK_HEADER_BYTE(read_buf.mem), str.size()) == 0);
+        __wt_buf_free(nullptr, &read_buf);
+    }
+}
+
+void
+create_write_buffer(WT_BM *bm, std::shared_ptr<mock_session> session, std::string contents,
+  WT_ITEM *buf, size_t buf_memsize)
+{
+    // Fetch write buffer size from block manager.
+    REQUIRE(bm->write_size(bm, session->get_wt_session_impl(), &buf_memsize) == 0);
+    test_and_validate_write_size(bm, session, buf_memsize);
+
+    // Initialize the buffer with aligned size.
+    F_SET(buf, WT_ITEM_ALIGNED);
+    REQUIRE(__wt_buf_initsize(session->get_wt_session_impl(), buf, buf_memsize) == 0);
+
+    /*
+     * Copy content string into the buffer.
+     *
+     * Following the architecture guide, it seems that the block manager expects a block header. I
+     * have tried to mimic that here.
+     */
+    REQUIRE(__wt_buf_grow_worker(session->get_wt_session_impl(), buf, buf->size) == 0);
+    memcpy(WT_BLOCK_HEADER_BYTE(buf->mem), contents.c_str(), contents.length());
+    // REQUIRE(__wt_buf_set(session->get_wt_session_impl(), buf, contents.c_str(),
+    // contents.length()) == 0);
+}
+
+TEST_CASE("Block manager: file operation read, write and write_size functions", "[block_api]")
+{
+    // Build Mock session, this will automatically create a mock connection.
+    std::shared_ptr<mock_session> session = mock_session::build_test_mock_session();
+    config_parser cp({{"allocation_size", ALLOCATION_SIZE}, {"block_allocation", BLOCK_ALLOCATION},
+      {"os_cache_max", OS_CACHE_MAX}, {"os_cache_dirty_max", OS_CACHE_DIRTY_MAX},
+      {"access_pattern_hint", ACCESS_PATTERN}});
+
+    REQUIRE(
+      (session->get_mock_connection()->setup_block_manager(session->get_wt_session_impl())) == 0);
+    session->setup_block_manager_file_operations();
+
+    WT_BM *bm;
+    REQUIRE(__wt_blkcache_open(session->get_wt_session_impl(), "file:test", cp.get_config_array(),
```

Comment:
Calling `wt_blkcache_open` cannot be the right way of doing this test. That is testing the block cache and not the block manager.